sdk/src/azure/platform/az_curl.c

```diff
@@ -233,9 +233,12 @@ _az_http_client_curl_build_headers(az_http_request const* request, struct curl_s
 static AZ_NODISCARD az_result
 _az_http_client_curl_append_url(az_span writable_buffer, az_span url_from_request)
 {
+  /*
+  TODO: url-encode query parameters only. Not all the url
   int32_t length;
   AZ_RETURN_IF_FAILED(_az_span_url_encode(writable_buffer, url_from_request, &length));
-  az_span remainder = az_span_slice_to_end(writable_buffer, length);
+   */
+  az_span remainder = az_span_copy(writable_buffer, url_from_request);
```

Comment:
Shouldn't this go back to what it was previously? You are missing a length check now.

https://github.com/Azure/azure-sdk-for-c/pull/1002/files?file-filters%5B%5D=.c#diff-812fa33b8fd2da4df243b3a9689a502dL248
```C#
int32_t required_length = az_span_size(url_from_request) + 1;
AZ_RETURN_IF_NOT_ENOUGH_SIZE(writable_buffer, required_length);
writable_buffer = az_span_copy(writable_buffer, url_from_request);
az_span_copy_u8(writable_buffer, 0);
```

See the file before your change in https://github.com/Azure/azure-sdk-for-c/pull/1002:
https://github.com/Azure/azure-sdk-for-c/blob/e40ebd86f3ed64c333b2c6f6b73a6270b08f7ec0/sdk/src/azure/platform/az_curl.c#L248-L255

Reply:
This is partially why I suggested a clean revert (and re-apply the changes you need). But I understand your point about needing to keep parts of it. 

**Just make sure that the changes to `az_curl.c` match what we expect.**

Reply:
it was not really required to check the size in there.
The function `_az_http_client_curl_append_url` is only called by `_az_http_client_curl_setup_url` which allocates the size of the url + 1.

The reason it was there is just in case in the future someone else calls that method from someone else

Since it is just a MACRO.  I can just added again.

thanks for noticing